### PR TITLE
feat: implement support for atomic storage

### LINKED ISSUE #60: Integrate Atomic Storage Changes In SDK
Integrate Atomic Storage In SDK

This is enhancement atomic storage support optimized reading of state. https://github.com/sarvalabs/go-pisa/issues/3

### PR DESCRIPTION
**Pull request by sarvalabs-harshrastogi**
# Implement Atomic Storage Support

This pull request introduces a breaking change that modifies how developers interact with the persistence state using SDK. The proposed approach aims to enhance read performance, particularly for complex data structures like arrays, variable arrays, maps, and classes.

This implementation streamlines data retrieval by leveraging atomic storage, resulting in improved efficiency and responsiveness when handling intricate field types.

* closes #60 

## Changes include
- [ ] Bugfix (non-breaking change that solves an issue)
- [X] New feature (non-breaking change that adds functionality)
- [X] Breaking change (change that is not backwards-compatible and/or changes current functionality)

## Checklist
- [X] I have assigned this PR to myself
- [X] I have added at least 1 reviewer
- [X] I have tested this code
- [X] I have updated the README and other relevant documents (guides...)

### CODE REVIEW

**Review by sarvalabs-gokul — CHANGES_REQUESTED** [review]

File: docs/source/logic.rst [hunk 1]
```diff
@@ -204,12 +204,34 @@ Variables
...
 different invocations and interactions.
 
-``ephemeralState`` - The ephemeral state, accessible via this variable, 
-represents the short-term or temporary state of the logic.
+ It contains the following method:
+
+* ``get`` 
+    This method retrieves the value from persistent state using the storage key.
+    As the storage key hash generation is complex, we have exposed the a builder 
+    object to generate the storage key hash. The builder object has the following methods:
```
[the remark this answers is not in the document]
Comment (sarvalabs-harshrastogi) [reply docs/source/logic.rst]: 6c5d330aae58dc039e7037f8931560ab53d3334b

File: docs/source/manifest-call-encoder.rst [hunk 2]
```diff
@@ -229,13 +229,21 @@ Methods
...
-    >> { class: 'string', data: 'Something went wrong', trace: [ 'function1()', 'function2()' ] }
+    >> {
+            Class: "string",
+            Error: "insufficient balance for sender",
+            Revert: false,
+            Trace: [
+                "runtime.root()", 
+                "routine.Transfer() [0xc] ... [0x1b: THROW 0x5]"
+            ],
+        }
```
[the remark this answers is not in the document]
Comment (sarvalabs-harshrastogi) [reply docs/source/manifest-call-encoder.rst]: 6c5d330aae58dc039e7037f8931560ab53d3334b

File: packages/js-moi-logic/__tests__/logic.test.ts [hunk 3]
```diff
@@ -68,59 +69,47 @@ describe("Logic", () => {
...
-                expect(logicId).toBeDefined();
-                return;
-            };
+            if (logicId == null) {
+                const factory = new LogicFactory(manifest, wallet);
+                const ix = await factory.deploy("Seeder", SYMBOL, INITIAL_SUPPLY);
+                await ix.wait();
+                const result = await ix.result();
+                logicId = result.logic_id;
+            }
```
[the remark this answers is not in the document]
Comment (sarvalabs-harshrastogi) [reply packages/js-moi-logic/__tests__/logic.test.ts]: a9f51bf698c415bfb97c9b13382b04de873895b9

File: packages/js-moi-manifest/types/manifest.d.ts [hunk 4]
```diff
@@ -22,7 +22,8 @@ export declare module LogicManifest {
     }
     
     export interface State {
-        kind: string;
+        kind?: string;
+        mode?: string;
```
[the remark this answers is not in the document]
Comment (sarvalabs-harshrastogi) [reply packages/js-moi-manifest/types/manifest.d.ts]: a9f51bf698c415bfb97c9b13382b04de873895b9

File: packages/js-moi-logic/src.ts/state/entity-builder.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        this.slotAccessorBuilder = new SlotAccessorBuilder(field, this.elementDescriptor);
+        return this.slotAccessorBuilder;
+    }
+
+    getSlotAccessorBuilder(): AccessorBuilder {
+        if (this.slotAccessorBuilder == null) {
+            ErrorUtils.throwError("Slot accessor builder not initialized");
+        }
+
+        return this.slotAccessorBuilder;
```
[the remark this answers is not in the document]
Comment (sarvalabs-harshrastogi) [reply packages/js-moi-logic/src.ts/state/entity-builder.ts:43]: a9f51bf698c415bfb97c9b13382b04de873895b9

**Review by sarvalabs-gokul — APPROVED** [review]

**Pull request merged**